source/DomainTheory/Topology/ScottTopologyProperties.lagda

```diff
@@ -0,0 +1,223 @@
+
+open import MLTT.Spartan
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.SubtypeClassifier
+
+module DomainTheory.Topology.ScottTopologyProperties
+        (pt : propositional-truncations-exist)
+        (fe : Fun-Ext)
+        (𝓥  : Universe) where
+
+open import UF.Logic
+open Existential pt
+open Implication fe
+open Universal   fe
+open Conjunction
+
+open import UF.Size
+open import UF.Equiv
+
+open import UF.Powerset-MultiUniverse
+open import Slice.Family
+
+open PropositionalTruncation pt
+
+open import DomainTheory.Topology.ScottTopology        pt fe 𝓥
+open import DomainTheory.Basics.Dcpo                   pt fe 𝓥
+open import DomainTheory.BasesAndContinuity.Continuity pt fe 𝓥
+open import DomainTheory.Basics.WayBelow               pt fe 𝓥
+
+\end{code}
+
+\begin{code}
+
+principal-filter : (𝓓 : DCPO {𝓤} {𝓥}) → ⟨ 𝓓 ⟩ → 𝓟 ⟨ 𝓓 ⟩
+principal-filter 𝓓 c x = c ⊑⟨ 𝓓 ⟩ x , prop-valuedness 𝓓 c x
+
+infix 45 principal-filter
+
+syntax principal-filter 𝓓 x = ↑[ 𝓓 ] x
+
+\end{code}
+
+Let `D` be a dcpo and consider a compact element `c : D` of it. The
+upwards-closure of `c` is then a Scott open.
+
+\begin{code}
+
+module Properties (𝓓 : DCPO {𝓤} {𝓥}) where
+
+ open DefnOfScottTopology 𝓓 𝓥
+
+ principal-filter-is-upwards-closed : (x : ⟨ 𝓓 ⟩)
+                                    → is-upwards-closed (↑[ 𝓓 ] x) holds
+ principal-filter-is-upwards-closed x y z p q =
+  x ⊑⟨ 𝓓 ⟩[ p ] y ⊑⟨ 𝓓 ⟩[ q ] z ∎⟨ 𝓓 ⟩
+
+ compact-implies-principal-filter-is-scott-open : (c : ⟨ 𝓓 ⟩)
+                                                → is-compact 𝓓 c
+                                                → is-scott-open (↑[ 𝓓 ] c) holds
+ compact-implies-principal-filter-is-scott-open c κ = Ⅰ , Ⅱ
+  where
+   Ⅰ : is-upwards-closed (↑[ 𝓓 ] c) holds
+   Ⅰ = principal-filter-is-upwards-closed c
+
+   Ⅱ : is-inaccessible-by-directed-joins (↑[ 𝓓 ] c) holds
+   Ⅱ (S , δ) = κ (index S) (S [_]) δ
+
+\end{code}
+
+Conversely, if the principal filter is Scott open then `c` is a compact element.
+
+\begin{code}
+
+ principal-filter-scott-open-implies-compact : (c : ⟨ 𝓓 ⟩)
+                                             → is-scott-open (↑[ 𝓓 ] c) holds
+                                             → is-compact 𝓓 c
+ principal-filter-scott-open-implies-compact c (υ , κ) I ι δ p =
+  κ ((I , ι) , δ) p
+
+\end{code}
+
+We can now record this as a logical equivalence.
+
+\begin{code}
+
+ principal-filter-scott-open-iff-compact :
+  (x : ⟨ 𝓓 ⟩) → is-scott-open (↑[ 𝓓 ] x) holds ↔ is-compact 𝓓 x
+ principal-filter-scott-open-iff-compact x = Ⅰ , Ⅱ
+  where
+   Ⅰ = principal-filter-scott-open-implies-compact x
+   Ⅱ = compact-implies-principal-filter-is-scott-open x
+
+\end{code}
+
+Notation for the principal Scott open.
+
+\begin{code}
+
+ ↑ˢ[_] : (Σ c ꞉ ⟨ 𝓓 ⟩ , is-compact 𝓓 c) → Σ S ꞉ 𝓟 {𝓥} ⟨ 𝓓 ⟩ , is-scott-open S holds
+ ↑ˢ[ (c , κ) ] =
+  principal-filter 𝓓 c , compact-implies-principal-filter-is-scott-open c κ
+
+\end{code}
+
```

Comment:
Did you want to write a comment between the `\end{code}` and `\begin{code}`? This applies a few times below as well.

Reply:
I do this instinctively to mark separation between chunks of code that are not directly related to each other. Fixed it now (as it is kind of a nonstandard convention in `TypeTopology`).

Reply:
I do the same, but I also find it's usually a sign that some comments are warranted.